src/Plugin/Field/FieldFormatter/InstagramEmbedFormatter.php

```diff
@@ -34,10 +38,16 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
     foreach ($items as $delta => $item) {
       $matches = [];
       foreach (Instagram::$validationRegexp as $pattern => $key) {
-        if (preg_match($pattern, $item->value, $matches)) {
+        if (preg_match($pattern, $this->getEmbedCode($item), $item_matches)) {
+          $matches[] = $item_matches;
           break;
         }
       }
+
+      if (!empty($matches)) {
+        $matches = reset($matches);
```

Comment:
Ok, got it. If we skip the other matches then there is no 
```php
$matches['shortcode']
```
then we build no renderable array for a field value.
The shortcode is the 5th match (according to [phpliveregex](http://www.phpliveregex.com/)

![screen shot 2015-12-09 at 09 32 22](https://cloud.githubusercontent.com/assets/12140153/11680432/81b2021e-9e58-11e5-8751-016a1dac693a.png)